### PR TITLE
Remove VIA, VTC and PPC support

### PR DESCRIPTION
**Pull request by lvndry**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 

### 📝 Description

Remove VIA, VTC and PPC support

### ❓ Context

- **JIRA or GitHub link**: https://ledgerhq.atlassian.net/browse/LIVE-11926

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: libs/coin-modules/coin-bitcoin/src/signOperation.ts [hunk 1]
```diff
@@ -92,8 +89,8 @@ export const buildSignOperation =
             lockTime,
             sigHashType,
             segwit,
-            hasTimestamp,
-            initialTimestamp,
+            hasTimestamp: false,
+            initialTimestamp: undefined,
```
Comment (Wozacosta) [inline libs/coin-modules/coin-bitcoin/src/signOperation.ts:93]: does `initialTimestamp` ever get set anywhere now?

Comment (lvndry) [reply libs/coin-modules/coin-bitcoin/src/signOperation.ts:93]: No it doesn't
However I looked into when it was called and it's a bit of a rabbit hole so I just keep the changes limited

**Review by Wozacosta — DISMISSED** [review]

File: libs/coin-modules/coin-bitcoin/src/signOperation.ts (hunk 1 above)
Comment (Wozacosta) [reply libs/coin-modules/coin-bitcoin/src/signOperation.ts:93]: Yeah it could be a separate issue then, you're right

**Comment by hzheng-ledger** [comment]
1. Not directly related to the PR, Could you please remove the Stealth (XST) too? We don't support it since a longtime in LL?
2.  Could you please remove the code from ledger-live/libs/ledgerjs/packages/types-cryptoassets/src/index.ts too? (the "LedgerExplorerId" part, line 167. Remove xst and xsn) As we don't support them in terms of Ledger explorer.
3. Remove code from ledger-live/libs/ledger-live-common/src/appSupportsQuitApp.ts please

File: libs/ledgerjs/packages/types-cryptoassets/src/index.ts [hunk 2]
```diff
@@ -177,15 +176,10 @@ export type LedgerExplorerId =
   | "hsr"
   | "kmd"
   | "ltc"
-  | "ppc"
   | "pivx"
   | "posw"
   | "qtum"
-  | "xsn"
```
Comment (Wozacosta) [inline libs/ledgerjs/packages/types-cryptoassets/src/index.ts:187]: was that voluntary?

Comment (lvndry) [reply libs/ledgerjs/packages/types-cryptoassets/src/index.ts:187]: Yes
cf. https://github.com/LedgerHQ/ledger-live/pull/7138#issuecomment-2180117026

Comment (Wozacosta) [reply libs/ledgerjs/packages/types-cryptoassets/src/index.ts:187]: Ah yes, only read xst. 👍

File: libs/coin-modules/coin-bitcoin/src/wallet-btc/wallet.ts [hunk 3]
```diff
@@ -218,7 +218,7 @@ class BitcoinLikeWallet {
       onDeviceSignatureGranted,
       onDeviceStreaming,
     } = params;
-    let hasTimestamp = params.hasTimestamp;
+    const hasTimestamp = params.hasTimestamp;
```
Comment (hzheng-ledger) [inline libs/coin-modules/coin-bitcoin/src/wallet-btc/wallet.ts:221]: in fact, 
if I don't make mistake, all the "hasTimestamp" and the "initialTimestamp" logic can be removed from coin-btc module.
they are only used in peercoin.
We can remove these parameters in the related methods

File: libs/coin-modules/coin-bitcoin/src/signOperation.ts (hunk 1 above)
Comment (hzheng-ledger) [reply libs/coin-modules/coin-bitcoin/src/signOperation.ts:93]: we can get rid of them. but of course, we can do it in another task if you want.

File: libs/coin-modules/coin-bitcoin/src/wallet-btc/wallet.ts (hunk 3 above)
Comment (hzheng-ledger) [reply libs/coin-modules/coin-bitcoin/src/wallet-btc/wallet.ts:221]: we can do it in another ticket if you want

Comment (Wozacosta) [reply libs/coin-modules/coin-bitcoin/src/wallet-btc/wallet.ts:221]: I'll jump in and do it on this PR @hzheng-ledger

File: libs/ledgerjs/packages/cryptoassets/src/currencies.ts [hunk 4]
```diff
@@ -644,7 +644,6 @@ export const cryptocurrenciesById: Record<CryptoCurrencyId, CryptoCurrency> = {
       P2PKH: 28,
       P2SH: 85,
       XPUBVersion: 0x0488b21e,
-      hasTimestamp: true,
```
Comment (Wozacosta) [inline libs/ledgerjs/packages/cryptoassets/src/currencies.ts:647]: I've only removed the variable here, even though those currencies are not supported by LL anymore, do you suggest I also remove those or should they be kept here @hzheng-ledger ?

**Review by lvndry — COMMENTED** [review]
LGTM

**Review by hzheng-ledger — DISMISSED** [review]
looks good. Thanks a lot!

File: libs/ledgerjs/packages/cryptoassets/src/currencies.ts (hunk 4 above)
Comment (hzheng-ledger) [reply libs/ledgerjs/packages/cryptoassets/src/currencies.ts:647]: good catch. @Wozacosta I think we can remove it too to simplify our codebase. thanks a lot

**Review by hzheng-ledger — APPROVED** [review]
good job, thanks a lot

**Review by cgrellard-ledger — APPROVED** [review]

**Pull request merged**